File: fronted/src/services/supplier.js
import api from './api'

// 供应商服务
export default {
  // 获取供应商列表
  getSuppliers(params = {}) {
    return api.get('/api/v1/suppliers', { params })
  },

  // 获取单个供应商
  getSupplier(id) {
    return api.get(`/api/v1/suppliers/${id}`)
  },

  // 创建供应商
  createSupplier(supplierData) {
    // 转换字段名称以匹配后端API
    const apiData = {
      name: supplierData.name,
      code: supplierData.code,
      type: supplierData.type,
      contact_person: supplierData.contactPerson,
      contact_phone: supplierData.contactPhone,
      email: supplierData.email,
      address: supplierData.address,
      city: supplierData.city,
      rating: supplierData.rating,
      qualification: supplierData.qualification,
      payment_terms: supplierData.paymentTerms,
      delivery_terms: supplierData.deliveryTerms,
      status: supplierData.status,
      note: supplierData.note
    }

    console.log('发送到后端的供应商数据:', apiData)
    return api.post('/api/v1/suppliers', apiData)
  },

  // 更新供应商
  updateSupplier(id, supplierData) {
    // 转换字段名称以匹配后端API
    const apiData = {
      name: supplierData.name,
      code: supplierData.code,
      type: supplierData.type,
      contact_person: supplierData.contactPerson,
      contact_phone: supplierData.contactPhone,
      email: supplierData.email,
      address: supplierData.address,
      city: supplierData.city,
      rating: supplierData.rating,
      qualification: supplierData.qualification,
      payment_terms: supplierData.paymentTerms,
      delivery_terms: supplierData.deliveryTerms,
      status: supplierData.status,
      note: supplierData.note
    }

    console.log('更新供应商数据:', apiData)
    return api.put(`/api/v1/suppliers/${id}`, apiData)
  },

  // 删除供应商
  deleteSupplier(id) {
    return api.delete(`/api/v1/suppliers/${id}`)
  }
}

File: mobile/lib/models/supplier.dart
import 'package:json_annotation/json_annotation.dart';

part 'supplier.g.dart';

enum SupplierType {
  @JsonValue('manufacturer')
  manufacturer,
  @JsonValue('distributor')
  distributor,
  @JsonValue('retailer')
  retailer,
  @JsonValue('wholesaler')
  wholesaler,
  @JsonValue('other')
  other,
}

extension SupplierTypeExtension on SupplierType {
  String toJson() {
    switch (this) {
      case SupplierType.manufacturer:
        return 'manufacturer';
      case SupplierType.distributor:
        return 'distributor';
      case SupplierType.retailer:
        return 'retailer';
      case SupplierType.wholesaler:
        return 'wholesaler';
      case SupplierType.other:
        return 'other';
    }
  }
}

enum SupplierRating {
  @JsonValue('S')
  S,
  @JsonValue('A')
  A,
  @JsonValue('B')
  B,
  @JsonValue('C')
  C,
  @JsonValue('D')
  D,
}

@JsonSerializable()
class Supplier {
  @JsonKey(name: 'ID')
  final int id;
  @JsonKey(name: 'Name')
  final String name;
  @JsonKey(name: 'Code')
  final String code;
  @JsonKey(name: 'Type')
  final SupplierType type;
  @JsonKey(name: 'ContactPerson')
  final String? contactPerson;
  @JsonKey(name: 'ContactPhone')
  final String? contactPhone;
  @JsonKey(name: 'Email')
  final String? email;
  @JsonKey(name: 'Address')
  final String? address;
  @JsonKey(name: 'City')
  final String? city;
  @JsonKey(name: 'Rating')
  final SupplierRating? rating;
  @JsonKey(name: 'Qualification')
  final String? qualification;
  @JsonKey(name: 'PaymentTerms')
  final String? paymentTerms;
  @JsonKey(name: 'DeliveryTerms')
  final String? deliveryTerms;
  @JsonKey(name: 'Status')
  final bool status;
  @JsonKey(name: 'Note')
  final String? note;
  @JsonKey(name: 'CreatedAt')
  final DateTime createdAt;
  @JsonKey(name: 'UpdatedAt')
  final DateTime updatedAt;

  Supplier({
    required this.id,
    required this.name,
    required this.code,
    required this.type,
    this.contactPerson,
    this.contactPhone,
    this.email,
    this.address,
    this.city,
    this.rating,
    this.qualification,
    this.paymentTerms,
    this.deliveryTerms,
    required this.status,
    this.note,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Supplier.fromJson(Map<String, dynamic> json) =>
      _$SupplierFromJson(json);

  Map<String, dynamic> toJson() => _$SupplierToJson(this);
}
